query/src/test/java/com/connexta/replication/query/cmp/QueryServiceComponentTest.java

```diff
@@ -0,0 +1,421 @@
+          collector.addError(t);
+          throw t;
+        };
+  }
+
+  @After
+  public void cleanup() throws Exception {
+    tasks.clear();
+    filterManager.objects().map(Filter::getId).forEach(filterManager::remove);
+    siteManager.objects().map(Site::getId).forEach(siteManager::remove);
+  }
+
+  /*
+   Verifies that multiple metacards can be handled and that the proper date is saved in the filterIndex
+  */
+  @Test
+  public void testQueryMultipleMetacards() throws Exception {
+    final Site site = new SiteImpl(generateSitePojo(1));
+    siteManager.save(site);
+    final Filter filter = new FilterImpl(generateFilterPojo("1", 11));
+    filterManager.save(filter);
+    List<Metadata> metadataList =
+        List.of(
+            generateMetadataWithResource(111),
+            generateMetadataWithResource(112),
+            generateMetadataWithResource(113, new Date(1)));
+    when(adapter.query(Mockito.argThat(isQueryRequestWithFilter(filter.getFilter()))))
+        .thenReturn(
+            new QueryResponseImpl(
+                metadataList,
+                e -> {
+                  throw new AssertionError(e);
+                }));
+
+    indexManager = Mockito.spy(indexManager);
+
+    startQueryManager(THIRTY_MINUTES);
+
+    final MemorySiteQueue queue = (MemorySiteQueue) broker.getQueue(site.getId());
+
+    assertThat(
+        "timed out waiting for expected published task",
+        queue.waitForPendingSizeToReach(3, 5L, TimeUnit.SECONDS),
+        Matchers.equalTo(true));
+
+    for (Metadata metadata : metadataList) {
+      assertTask(queue.take(), metadata, (byte) 0, OperationType.HARVEST);
+    }
+    verify(indexManager)
+        .save(
+            Mockito.argThat(
+                isIndexWithModified(metadataList.get(2).getMetadataModified().toInstant())));
+  }
+
+  @Test
+  public void testQueryThrough2Polls() throws Exception {
+    final Site site = new SiteImpl(generateSitePojo(1));
+    siteManager.save(site);
+    final Filter filter = new FilterImpl(generateFilterPojo("1", 11));
+    filterManager.save(filter);
+    List<Metadata> metadataList =
+        List.of(
+            generateMetadataWithResource(111),
+            generateMetadataWithResource(112),
+            generateMetadataWithResource(113, new Date(1)));
+    Metadata secondPollMetadata = generateMetadataWithResource(114, new Date(2));
+    when(adapter.query(Mockito.argThat(isQueryRequestWithFilter(filter.getFilter()))))
+        .thenReturn(
+            new QueryResponseImpl(
+                metadataList,
+                e -> {
+                  throw new AssertionError(e);
+                }),
+            new QueryResponseImpl(
+                List.of(secondPollMetadata),
+                e -> {
+                  throw new AssertionError(e);
+                }));
+
+    indexManager = Mockito.spy(indexManager);
+
+    startQueryManager(THIRTY_MINUTES);
```

Comment:
❓ Wouldn't this make this test take 30 minutes to run? We need to scale time to 1 ms (i.e. the minimum) and introduce ways to ensure the test always run the same way.

Reply:
This is the config polling period. Not the query polling period